### PR TITLE
Storage requester pays sample code + tests

### PR DESCRIPTION
**Pull request by ryanmats**
Code samples + tests are complete and ready for review.

HOWEVER, we should not merge this until a bug in the client library is resolved (https://github.com/GoogleCloudPlatform/google-cloud-php/issues/667). The bug prevents the download function from working properly when using different accounts for the Storage owner and the requester.

### CODE REVIEW

File: storage/api/src/functions/disable_requester_pays.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ */
+function disable_requester_pays($projectId, $bucketName)
+{
+    $storage = new StorageClient([
+        'projectId' => $projectId
+    ]);
+    $bucket = $storage->bucket($bucketName);
+    $bucket->update([
+        'billing' => [
+            'requesterPays' => False
```
Comment (bshaffer) [inline storage/api/src/functions/disable_requester_pays.php]: `False` should be lowercase

File: storage/api/src/functions/enable_requester_pays.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ */
+function enable_requester_pays($projectId, $bucketName)
+{
+    $storage = new StorageClient([
+        'projectId' => $projectId
+    ]);
+    $bucket = $storage->bucket($bucketName);
+    $bucket->update([
+        'billing' => [
+            'requesterPays' => True
```
Comment (bshaffer) [inline storage/api/src/functions/enable_requester_pays.php]: again here

File: storage/api/test/RequesterPaysCommandTest.php [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+use Google\Cloud\Samples\Storage\RequesterPaysCommand;
+use Google\Cloud\Storage\StorageClient;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Tester\CommandTester;
+
+/**
+ * Unit Tests for RequesterPaysCommand.
+ */
+class BucketsCommandTest extends \PHPUnit_Framework_TestCase
```
Comment (bshaffer) [inline storage/api/test/RequesterPaysCommandTest.php]: should be `RequesterPaysCommandTest`

File: storage/api/test/RequesterPaysCommandTest.php [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                'bucket' => $bucketName,
+                '--enable' => true,
+            ],
+            ['interactive' => false]
+        );
+
+        $this->expectOutputRegex("/Requester pays has been enabled/");
+    }
+
+    public function testDisableRequesterPays()
```
Comment (bshaffer) [inline storage/api/test/RequesterPaysCommandTest.php]: This should have `/** @depends testEnableRequesterPays */` so it only runs if the above test passes

File: storage/api/test/RequesterPaysCommandTest.php [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                'bucket' => $bucketName,
+                '--disable' => true,
+            ],
+            ['interactive' => false]
+        );
+
+        $this->expectOutputRegex("/Requester pays has been disabled/");
+    }
+
+    public function testGetRequesterPaysStatus()
```
Comment (bshaffer) [inline storage/api/test/RequesterPaysCommandTest.php]: This should have `/** @depends testDisableRequesterPays */` so it only runs if the above test passes

File: storage/api/test/RequesterPaysCommandTest.php [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+                'project' => $projectId,
+                'bucket' => $bucketName,
+                'object' => $objectName,
+                'download-to' => $destination
+            ],
+            ['interactive' => false]
+        );
+
+        $this->expectOutputRegex("/Downloaded/");
+        $this->expectOutputRegex("/using requester-pays requests/");
```
Comment (bshaffer) [inline storage/api/test/RequesterPaysCommandTest.php]: The `expectOutputRegex` function only uses the last value its called with, so `"/Downloaded/"` should be removed.

File: storage/api/src/functions/download_file_requester_pays.php [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+ * @param string $destination Path to local file to save.
+ *
+ * @return void
+ */
+function download_file_requester_pays($projectId, $bucketName, $objectName, $destination)
+{
+    $storage = new StorageClient([
+        'projectId' => $projectId
+    ]);
+    $bucket = $storage->bucket($bucketName, true);
```
Comment (bshaffer) [inline storage/api/src/functions/download_file_requester_pays.php]: We should make this example more obvious as to how this applies to Requester Pays, like this:

```php
$storage = new StorageClient([
    'projectId' => $projectId
]);
$requesterPays = true;
$bucket = $storage->bucket($bucketName, $requesterPays);
```

**Comment by bshaffer** [comment]
Hmm, the 5.6 build errored with "Your requirements could not be resolved to an installable set of packages". I restarted the build because I've never seen this error before. 🤞

**Comment by frankyn** [comment]
Pending open issue. 

Sample design LGTM!

**Comment by ryanmats** [comment]
It could be an issue where I've been testing everything using PHP version 7.1 on my computer, and that updates the composer files to use something that is unusable in version 5.6.

If this build fails again I can try moving to 5.6 and push any updates to the composer files.

**Pull request merged**